### ISSUE TITLE
runner.go no longer correctly parses arguments for subprocess (i.e. envconsul can no longer spawn suprocesses that require arguments)

### ISSUE THREAD

**Issue by avislash** [hashicorp/envconsul#297]
runner.go as of main commit [6b2eef8](https://github.com/hashicorp/envconsul/commit/6b2eef8bef204947ac9f75be7657f6a19a05f683) no longer correctly parses commands with arguments. 

The behavior shown now when envoking envconsul with the following CLI arguments 

```
envconsul -log-level=info --no-prefix=true -secret="secret/path" "./bin/foo arg1"
```

Will try to spawn a child process for an executable called "/bin/foo arg1" instead of spawning a child process using the executable "/bin/foo" with arg1 passed to it.

In examining the diff between this commit and the previous commit it looks like the shell parser was removed in runner.go commit [2c2c9be](https://github.com/hashicorp/envconsul/commit/2c2c9beee42bd2380faf68edcc00d32eb92294dd#diff-0ccec778a288e27b5a6ada45110d2a1855585629224a3dd4a8b5ff6f9319e283) 

and now the raw string is passed in as the command with no seperating between command and arguments.

**Comment by eikenb** [comment]
Hey @avislash, thanks for the bug report.

I'm guessing this is on the current HEAD. That included an update to the consul-template version it uses under the hood which reworked how it handled commands. Eliminating the shell parsing as the shell parsing library was riddled with problems.

It now takes either a string that should get run with something like `sh -c 'command with args'` so the shell handles the parsing or it takes a list `["command", "arg", "arg"]` that Go can deal with directly.

I shot for backwards compatibility by having most cases in envconsul use the `sh -c` mode but there might have been some gaps. I'll look into this before releasing.

Thanks a ton for catching this!! Super appreciate people testing head as the tests have holes. :100:

**Comment by eikenb** [comment]
What you have for the command should work, that it doesn't is a bug.

Release wise we use main/HEAD as the development branch for the next release and try to keep it stable (ready to run) but as it is all the new code there will be more bugs. For stable release either grab a binary from [here](https://releases.hashicorp.com/envconsul/0.12.1) or you can checkout the stable branch (`git checkout v0.12.1`) and build it yourself.

**Comment by avislash** [comment]
> What you have for the command should work, that it doesn't is a bug.
> 
> Release wise we use main/HEAD as the development branch for the next release and try to keep it stable (ready to run) but as it is all the new code there will be more bugs. For stable release either grab a binary from [here](https://releases.hashicorp.com/envconsul/0.12.1) or you can checkout the stable branch (`git checkout v0.12.1`) and build it yourself.

Is there any reason to prefer checking out a tagged version and building it versus using
```
go install github.com/hashicorp/envconsul@latest
```

**Comment by eikenb** [comment]
I didn't think about go install... but yeah, that should be equivalent to checking it out locally and building the latest tagged release (0.12.1). Do you have the issue with 0.12.1?

Just to clarify. I do think this is a bug in HEAD and I'm hoping you have better luck with 0.12.1 until I get it fixed in HEAD.

Thanks for all the feedback!

**Comment by eikenb** [comment]
Forgot to tag this in the PR for the fix.

This is fixed by #304
